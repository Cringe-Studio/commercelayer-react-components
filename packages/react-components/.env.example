# Unit tests
VITE_TEST_CLIENT_ID=
VITE_TEST_ENDPOINT=
VITE_TEST_MARKET_ID=
VITE_TEST_USERNAME=
VITE_TEST_PASSWORD=
VITE_TEST_CLIENT_ID_INTEGRATION=
VITE_TEST_CLIENT_SECRET=
