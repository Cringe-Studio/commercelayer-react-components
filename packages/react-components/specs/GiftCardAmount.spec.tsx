import React from 'react'
import { GiftCardAmount } from '../packages/react-components/src'
import renderer from 'react-test-renderer'
import components from '../#config/components'
import BaseOrderPrice from '../#components/utils/BaseOrderPrice'
import Parent from '../#components/utils/Parent'

const propTypes = components.GiftCardAmount.propTypes

test('<GiftCardAmount/>', () => {
  expect.assertions(3)
  const component = renderer.create(<GiftCardAmount />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.format).toBe(propTypes.format)
})

test('<GiftCardAmount check children format />', () => {
  expect.assertions(4)
  const component = renderer.create(<GiftCardAmount format="float" />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.type).toBe(BaseOrderPrice)
  expect(rendered.props.format).toBe('float')
  expect(childRendered.type).toBe('span')
})

test('<GiftCardAmount with custom children />', () => {
  expect.assertions(4)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <GiftCardAmount>{CustomComponent}</GiftCardAmount>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.props.children).toBe(CustomComponent)
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(Parent)
})
