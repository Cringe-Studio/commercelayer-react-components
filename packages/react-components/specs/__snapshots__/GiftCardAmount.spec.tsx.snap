// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftCardAmount check children format /> 1`] = `
<span>
  
</span>
`;

exports[`<GiftCardAmount with custom children /> 1`] = `<span />`;

exports[`<GiftCardAmount/> 1`] = `
<span>
  
</span>
`;
